### ISSUE TITLE
Slash per-subcommand permissions

### ISSUE THREAD

**Issue by timjuic** [discordjs/discord.js#8300]
### Which package is the feature request for?

discord.js

### Feature

Currently there is no way to set permissions for specific slash subcommands. They can only be set for the command as a whole using `.setDefaultMemberPermissions()` or passing a sole object. I have a large bot for which I decided recode all legacy commands into slash commands. I ran into this problem that I have commands with some subcommands that can be ran by everyone, and some subcommands that can only be ran by administrators or require some other permission. When I started with slash commands I first thought I would be able to set required permissions for each subcommand, but that is apparently not the case. I know that I can make my own handler to handle subcommand permissions, but the thing is, when a user doesnt have required permissions for a slash command, it is hidden from the user when he types `/`. And thats what I want to achieve for subcommands. I want a user to see only subcommands that he has access to, other subcommands should be hidden from them. 

I got an idea of making all commands without subcommands to bypass this. But obviously that is not how it should be and makes subcommands useless. Heres the example. Note that `suggestions-deny` , `suggestions-accept`  and `suggestions-setup` are hidden to regular users because these are admin commands.
![Screenshot 2022-07-17 at 16 25 23](https://user-images.githubusercontent.com/70685646/179403446-35189f07-82b7-438e-87ef-554014c1a4b7.png)

### Ideal solution or implementation

So to sum it up, I am wondering if assigning required permissions for individual slash subcommands will be possible. Are you planning to add that feature or will you consider it if you havent thought about it yet? 
![Screenshot 2022-07-17 at 16 41 32](https://user-images.githubusercontent.com/70685646/179403620-a02337b1-b0fb-4d37-a976-7424ec25144a.png)

**Comment by Jiralite** [comment]
This is discord.js. We are not Discord. You suggest feature requests to Discord and then they are implemented in third-party libraries, such as discord.js (us).
